java-generator/it/src/it/ignore-unknown-toplevel-objects/src/test/java/io/fabric8/it/top/TestIgnoreUnknownTopleveObjsFromConfig.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.it.top;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.kubernetes.api.model.AnyType;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.Test;
+import io.fabric8.java.generator.testing.KubernetesResourceDiff;
+import io.argoproj.v1alpha1.Application;
+import io.argoproj.v1alpha1.ApplicationSpec;
+import io.argoproj.v1alpha1.applicationspec.Destination;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class TestIgnoreUnknownTopleveObjsFromConfig {
+
+  @Test
+  void testDeserialization() {
+    try {
+      Application.class.getMethod("getAdditionalProperties");
+      fail("No getAdditionalProperties method expected");
+		} catch (NoSuchMethodException e) {
```

Comment:
can you rebase on latest `main` spotless should now be applied also to those files.